src/tests/lib/repositories_v2/sqlalchemy_mock_builder.py

```diff
@@ -0,0 +1,127 @@
+from unittest import mock
+from src.tests.lib.repositories_v2.generic_sqlalchemy_mock_builder import (
+    GenericQueryMockBuilder,
+)
+
+
+class QueryMockBuilder(GenericQueryMockBuilder):
```

Comment:
we definitely need to refactor this.

Reply:
Maybe you can start with creating a data structure to handle the data?

Maybe a mix between a map and a linked list could help?